Source/JavaScriptCore/runtime/JSFunction.cpp

```diff
@@ -535,13 +546,13 @@ void JSFunction::setFunctionName(JSGlobalObject* globalObject, JSValue value)
         if (uid.isNullSymbol())
             name = emptyString();
         else
-            name = makeString('[', String(&uid), ']');
+            name = makeNameWithOutOfMemoryCheck(globalObject, scope, "Function ", '[', String(&uid), ']');
     } else {
         ASSERT(value.isString());
         name = asString(value)->value(globalObject);
         RETURN_IF_EXCEPTION(scope, void());
     }
-    reifyName(vm, globalObject, name);
+    RELEASE_AND_RETURN(scope, reifyName(vm, globalObject, name));
```

Comment:
If you make `reifyName` return `PropertyStatus`, you'll have to add a `(void)` cast to placate the compiler.